src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -171,7 +172,40 @@
  * {@link Builder#inheritInheritableThreadLocals(boolean) inheritInheritableThreadLocals}
  * method can be used to select if the initial values are inherited.
  *
- * <p> Unless otherwise specified, passing a {@code null} argument to a constructor
+ * <h2><a id="thread-interruption">Thread Interruption</a></h2>
+ * A {@code Thread} has an <em>interrupted status</em> which serves as a "request" for
+ * code executing in the thread to "stop or cancel its current activity". The interrupted
+ * status is set by invoking the target thread's {@link #interrupt()} method. Many methods
+ * that cause a thread to block or wait are <em>interruptible</em>, meaning they detect
+ * that the thread's interrupted status is set and cause execution to return early from
+ * the method, usually by throwing an exception.
+ *
+ * <p> If a thread executing {@link #sleep(long) Thread.sleep} or {@link Object#wait()
+ * Object.wait} is interrupted then it causes the method to return early and throw
+ * {@link InterruptedException}. Methods that throw {@code InterruptedException} do so after
```

Comment:
I think something like this is clearer:

```suggestion
 * Object.wait} is interrupted then it causes the method to exit early by throwing an
 * {@link InterruptedException}. Methods that throw {@code InterruptedException} do so after
```

Reply:
The previous paragraph establishes the phrase "return early" so this is why it is re-used here maybe "method to exit" would be better here, I'll play with it.